### DISCUSSION TITLE
Can I list files?

### DISCUSSION THREAD

**Discussion by naimur20** [Glitchii/tempfile.site#5] [Ideas]
Can I list files for same authorization keys using api? It would be very helpful.

**Comment by Glitchii** [comment]
You want to be able to list files which use the same auth key?
What if someone uploaded a file and used the same auth key as someone else?
This means they would be able to view files that aren't theirs 🤔.

**Comment by naimur20** [comment]
Perfectly said! 
Then you can provide systematically generated unique auth keys, thus user will not be able to use there own keys.

**Comment by Glitchii** [comment]
TLDR: No.

When uploading a file, a 3-word key is auto-generated. The user could choose to use that key or choose to override it with their own.

Forcing the user to use an auto-generated key means:
- The user might not be able to remember their key making their file inaccessible.
- The program would have to loop through all file entries in the database matching password hashes to see if the current key is already used (not a very good idea).

Probably a better way to have a feature that allows listing files would be to have accounts, however, I want this website to be simple, just temporary file hosting (very susceptible to change mind in the future).

If you do however have other suggestions, feel free to throw them out, just make sure to use the discussion page (https://github.com/Glitchii/tempfile.site/discussions) instead of making a GitHub issue.

I will now convert this issue into a discussion.
